app/models/adviser.rb

```diff
@@ -95,6 +95,10 @@ def upcase_postcode
     postcode.upcase! if postcode.present?
   end
 
+  def upcase_reference_number
+    reference_number&.upcase!
```

Comment:
Like the use of the safe navigator operator which we could not use prior to the ruby upgrade!